### PR TITLE
Maintenance Mode Improvements

### PR DESCRIPTION
**Pull request by SchrodingersGat**
Follow up to https://github.com/inventree/InvenTree/pull/6422

Provides some enhancements to maintenance mode support.

- Force maintenance mode when running management commands which change database data
- Custom "migrate" command (which wraps db migrations in management mode)

### CODE REVIEW

File: InvenTree/common/models.py [hunk 1]
```diff
@@ -703,6 +703,9 @@ def set_setting(cls, key, value, change_user=None, create=True, **kwargs):
                     attempts=attempts - 1,
                     **kwargs,
                 )
+        except (OperationalError, ProgrammingError):
+            # Likely the DB is locked - not much we can do here
+            pass
```
Comment (matmair) [inline InvenTree/common/models.py:708]: I think we should at least log this

File: tasks.py [hunk 2]
```diff
@@ -369,10 +369,8 @@ def migrate(c):
     print('Running InvenTree database migrations...')
     print('========================================')
 
-    manage(c, 'makemigrations')
-    manage(c, 'migrate --noinput')
-    manage(c, 'migrate --run-syncdb')
```
Comment (matmair) [inline tasks.py:374]: Question: are you sure we do not issue the sync command anymore?

**Review by matmair — APPROVED** [review]
LGTM

File: tasks.py (hunk 2 above)
Comment (SchrodingersGat) [reply tasks.py:374]: Added it back in :)

**Pull request merged**